lib/modules/manager/npm/post-update/__snapshots__/npm.spec.ts.snap

```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`modules/manager/npm/post-update/npm generates lock files 1`] = `
+exports[`modules/manager/npm/post-update/npm deduplicates dependencies after installation with NPM <= 6 1`] = `
```

Comment:
The `generate lock files` test was run with `npmDedupe` post update option. I refactored the `generate lock files` test to run with default options. Additionally I added two new tests for `npmDedupe` post update option: One running without NPM version constraint, one with NPM 6.

It is expected that the old `generates lock files` test equals the `deduplicates dependencies after installation with NPM <= 6` test.

I can create another small PR, which is only doing the refactoring to existing tests if you prefer that. We could merge that before and see here only the changes related to adding additonal logic for NPM >= 7. Up to you.